src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -470,6 +470,7 @@ private function addRouterSection(ArrayNodeDefinition $rootNode)
                     ->children()
                         ->scalarNode('resource')->isRequired()->end()
                         ->scalarNode('type')->end()
+                        ->scalarNode('base_uri')->defaultNull()->end()
```

Comment:
Is "base_uri" a good name? In the HTTP client context it means something else. This is more a "default_uri" or a "fallback_uri", Right?

Reply:
`base_uri` made sense to me, kinda similar to `router.request_context.base_url`, but I'm fine changing.
Anyone else with an opinion on this? @javiereguiluz @weaverryan since you're quite involved in the general DX?

Reply:
Adding `->info('The base URI used to generate URLs in a non-HTTP context')` might be enough, but `default_uri` would make sense to me as as well, reflecting the fact it's used whenever there is not already a context in which it can be inferred.

Is there something else in the meaning of `base_uri` in HTTP client differing from here, though?

Reply:
info added thanks.

Reply:
If you configure the HTTP client with a `base_uri: https://example.com` and make a request to `/foo`, the result will be `https://example.com/foo`. 

If you configure a router context with `base_uri: https://example.com` and generate a path for route `/foo` you will get `/foo`. If you generate a URL you will get `https://current-domain.com/foo`. 

A router context does only make sense if we didn't have a domain from a request object.

Reply:
In a "web" context, does the TRUE domain take priority over this new `base_uri` config value?

If yes, then I like `default_base_uri`

If no, then I like `base_uri`.

Reply:
I agree with @Nyholm that `base_uri` seems to indicate something else than what's actually doing. I like the `fallback_uri`, as that's exactly what it does (imho): it's a fallback when no request information is available.

Reply:
@weaverryan: Yes.

Reply:
`default_uri` seems the best option to me. If there is no URI, like in Console, then it uses the default. If not, the default is overridden.

Reply:
updated